server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java

```diff
@@ -705,11 +717,59 @@ public void getObject(@PathVariable final String bucketName,
       response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, ANY);
       response.setDateHeader(HttpHeaders.LAST_MODIFIED, s3Object.getLastModified());
       addUserMetadata(response::addHeader, s3Object);
+      addOverrideHeaders(response, request.getQueryString());
 
       try (final OutputStream outputStream = response.getOutputStream()) {
         Files.copy(s3Object.getDataFile().toPath(), outputStream);
       }
     }
+
+  }
+
+  private void addOverrideHeaders(final HttpServletResponse response, final String query) {
+    if (query != null && !query.isEmpty()) {
+      Arrays.stream(query.split("&"))
+          .map(this::splitQueryParameter)
+          .forEach((h) -> addOverrideHeader(response, h.getKey(), h.getValue()));
+    }
+  }
+
+  private SimpleImmutableEntry<String, String> splitQueryParameter(final String it) {
+    final int idx = it.indexOf("=");
+    try {
+      final String key = URLDecoder.decode(idx > 0 ? it.substring(0, idx) : it, UTF_8.name());
```

Comment:
Let's refactor that for better readability. We have Apaches StringUtils on board so something like this may work
```
    if (StringUtils.countMatches(it, '=') == 1) {
      var key = StringUtils.substringBefore(it, "=");
      var val = StringUtils.substringAfter(it, "=");
      ...
    }

```